### PR TITLE
Marketplace: Refactor - Pull AccountLink up into Marketplace

### PR DESCRIPTION
**Pull request by zspencer**
The account link object was being made in the controller, and while that worked it was a mess.

This adjusts things so it's a little bit more order(ly) and places it in the model where it's "best."

Co-authored-by: Dalton <daltonrpruitt@users.noreply.github.com>

### CODE REVIEW

File: app/furniture/marketplace/marketplace.rb [hunk 1]
```diff
@@ -30,6 +30,46 @@ def delivery_fee_cents
...
+        }).tap do |account|
+          update(stripe_account: account.id)
+        end
+      else
+        Stripe::Account.retrieve(stripe_account, {
+          api_key: stripe_api_key
+        })
+      end
+
+      account_link = if account.details_submitted?
```
Comment (anaulin) [inline app/furniture/marketplace/marketplace.rb:46]: Am I reading this right, that the only difference between the two if/else branches is the `type`? If so, maybe we can shorten this to something like:
```ruby
account_link = Stripe::AccountLink.create(
  {
    account: stripe_account,
    refresh_url: refresh_url,
    return_url: return_url,
    type: account.details_submitted? ? "account_update" : "account_onboarding"
  },
  {
    api_key: stripe_api_key
  }
)
```

**Review by anaulin — APPROVED** [review]

File: app/furniture/marketplace/marketplace.rb (hunk 1 above)
Comment (zspencer) [reply app/furniture/marketplace/marketplace.rb]: Good catch!
